### ISSUE TITLE
Parallel implementation of postprocessing routines

### ISSUE THREAD

**Issue by neurodroid** [losonczylab/sima#149]
I've added parallel implementations to two of the segmentation postprocessing routines:

https://github.com/neurodroid/sima/commit/18d26c89

Speedup is not overly impressive (at most 30% for the full post-processing step with real-world data).

If you think this might be useful, I'd be happy to submit a pull request.

**Comment by pkaifosh** [comment]
Nice! Just to confirm, the original functionality is all preserved (it seems so based on a very quick look)? If so, yes, please do submit a pull request. Thanks!

**Comment by neurodroid** [comment]
Let me do some more thorough testing and I'll get back to you.

**Comment by neurodroid** [comment]
Had to make a small change to make it exactly equivalent:
https://github.com/neurodroid/sima/commit/00250f32
I could submit a pull request now, unless you have any suggestions regarding implementation, naming, or style

**Comment by pkaifosh** [comment]
One possible naming change would be to underscore the Parallel classes if they are only intended to be used within the module. However, if you see them as being potentially used outside the module, then you can leave them as is. Otherwise, I think it looks great and ready to be submitted as a pull request.

**Comment by pkaifosh** [comment]
One other thing would be to add a test cases that uses each of these parallel versions. I think we should already have test cases for the sequential versions, so you can just copy these and set n_processes=2

**Comment by neurodroid** [comment]
Couldn't find any test cases for the segmentation, so I've added one:
https://github.com/neurodroid/sima/commit/7985149f
Let me know if this is a good place and strategy for a postprocessing test case.

**Comment by neurodroid** [comment]
P.S. Would have preferred to use sima.segment.STICA, but sklearn.FastICA appears to draw random numbers and I couldn't find a way to reproducibly seed their RNG

**Comment by pkaifosh** [comment]
@neurodroid Looks like a nice test case! The best place to put it would be in the directory sima/segment/tests rather than sima/tests
